src/app-layer-detect-proto.c

```diff
@@ -319,7 +319,7 @@ static inline int PMGetProtoInspect(AppLayerProtoDetectThreadCtx *tctx,
  *  \param direction direction for the patterns
  *  \param pm_results[out] AppProto array of size ALPROTO_MAX */
 static AppProto AppLayerProtoDetectPMGetProto(AppLayerProtoDetectThreadCtx *tctx, Flow *f,
-        const uint8_t *buf, uint16_t buflen, uint8_t flags, AppProto *pm_results, bool *rflow)
+        const uint8_t *buf, uint32_t buflen, uint8_t flags, AppProto *pm_results, bool *rflow)
```

Comment:
I did not manage to get `buflen` bugger than `UINT16_MAX` here, with big TCP window, first out of order packet, then first packet, with both of them getting a bigger payload than UINT16_MAX...
Is is possible ?

Reply:
I think in theory it might be. Debug validation bug on to see if fuzzing finds a case?

Reply:
Trying that...

Reply:
So, CIFuzz found cases in next version ;-)